doc/adrs/XXX_support_sorting_filtering_on_table.adoc

```diff
@@ -0,0 +1,117 @@
+= ADR-XXX Support for Sorting and Filtering in Tables
+
+== Context
+
+With the introduction of Material React Table to display table data,
+to ensure efficient data management from the outset, we have decided to adopt a server-side approach for filtering and sorting data.
+
+== Decision
+
+=== Global Searching
+
+==== Material React Table
+
+To turn on server-side filtering (client-side by default), we must enable `manuelFiltering`.
+
+[source, typescript]
+----
+const [globalFilter, setGlobalFilter] = useState('');
+
+useEffect(() => {
+    // retrieve data filtered from backend
+    setData([...filteredData]);
+}, [globalFilter]);
+
+const table = useMaterialReactTable({
+  columns,
+  data, // this will already be filtered on the server
+  manualFiltering: true, //turn off client-side filtering
+  onGlobalFilterChange: setGlobalFilter, //hoist internal global state to your state
+  state: { globalFilter }, //pass in your own managed globalFilter state
+});
+----
+
+==== Backend
+
+The following variable needs to be passed to the row semantic candidates expression:
+* globalFilterValue: text for filtering, empty if no global filtering set
```

Comment:
The name of the variable may be updated, I'm not sure if "globalFilterValue" would be properly understood but it's a detail too